core/network/src/test/sync.rs

```diff
@@ -140,6 +140,94 @@ fn sync_from_two_peers_with_ancestry_search_works() {
 		.canon_equals_to(net.peer(1).client.backend().as_in_memory().blockchain()));
 }
 
+#[test]
+fn ancestry_search_works_when_common_is_hundred() {
```

Comment:
Maybe extract this into a function to avoid repeating across multiple tests (common_is_one and common_is_two below).

Reply:
Forgot to rewrite with `test_ancestor_search_when_common_is`?